schemas/gateway-schema.json

```diff
@@ -0,0 +1,216 @@
+{
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "$id": "https://zowe.org/schema/api-layer-config",
+    "title": "Gateway service configuration",
+    "description": "Properties that can be used to configure gateway service in a configuration file, such as zowe.yaml.",
+    "type": "object",
+    "additionalProperties": true,
+    "properties": {
+        "port": {
+            "$ref": "#/$defs/port",
+            "default": 7553
+        },
+        "apiml": {
+            "type": "object",
+            "description": "Zowe API ML specific properties.",
+            "additionalProperties": false,
+            "properties": {
+                "security": {
+                    "type": "object",
+                    "description": "Authentication and authorization options",
+                    "additionalProperties": false,
+                    "properties": {
+                        "auth": {
+                            "type": "object",
+                            "description": "Authentication against API ML.",
+                            "additionalProperties": false,
+                            "properties": {
+                                "provider": {
+                                    "type": "string",
+                                    "description": "Which provider should be used to verify user credentials.",
+                                    "enum": ["zosmf","saf"],
+                                    "default": "zosmf"
+                                },
+                                "zosmf": {
+                                    "type": "object",
+                                    "description": "When zosmf is set as authentication provider, this section can be used to provide more details.",
+                                    "additionalProperties": false,
+                                    "properties": {
+                                        "jwtAutoconfiguration": {
+                                            "type": "string",
```

Comment:
I think this can be enum with JWT, LTPA, or AUTO

Reply:
added enum